testcases/PowerNVDump.py

```diff
@@ -262,32 +264,42 @@ def verify_dump_file(self, boot_type=BootType.NORMAL, dump_place="local"):
+                #if user is not root, cannot copy/list vmcore files due to permission issue, because the owner of vmcore dir will be root.
+                if self.dump_server_user == 'root':
+                    self.c.run_command('scp -i %s -r %s@%s:/%s/%s /var/crash/' %
+                                  (self.rsa_path, self.dump_server_user, self.dump_server_ip, self.dump_path, self.crash_content[0]), timeout=1200)
             if self.distro == "ubuntu":
                 self.c.run_command("ls /var/crash/%s/dump*" %
                                    self.crash_content[0])
             else:
-                res = self.c.run_command("ls /var/crash/%s/vmcore*" %
-                                         self.crash_content[0])
-                paths = res[0].split()
-                file_names = [os.path.basename(path) for path in paths]
-                # Check if vmcore-dmesg-incomplete.txt is present in file_names
-                if "vmcore-dmesg-incomplete.txt" in file_names:
-                    raise OpTestError("kdump failed to create vmcore file")
+                if self.dump_server_user != 'root':
+                    res = self.c.run_command("ssh %s@%s -i %s ls %s/%s/vmcore*" %
+                            (self.dump_server_user, self.dump_server_ip, self.rsa_path, self.dump_path, self.crash_content[0]))
                 else:
-                    filtered_files = [f for f in file_names if f.startswith("vmcore") and not f == "vmcore-dmesg.txt"]
-                    if filtered_files:
-                        log.debug("vmcore file  %s exists in crash dir" % filtered_files)
-                    else:
+                    res = self.c.run_command("ls /var/crash/%s/vmcore*" %
+                                          self.crash_content[0])
+                    paths = res[0].split()
+                    file_names = [os.path.basename(path) for path in paths]
+                    # Check if vmcore-dmesg-incomplete.txt is present in file_names
+                    if "vmcore-dmesg-incomplete.txt" in file_names:
                         raise OpTestError("kdump failed to create vmcore file")
+                    else:
+                        filtered_files = [f for f in file_names if f.startswith("vmcore") and not f == "vmcore-dmesg.txt"]
+                        if filtered_files:
+                            log.debug("vmcore file  %s exists in crash dir" % filtered_files)
+                        else:
+                            raise OpTestError("kdump failed to create vmcore file")
             if boot_type == BootType.MPIPL:
                 self.c.run_command("ls /var/crash/%s/opalcore*" %
                                    self.crash_content[0])
         else:
             msg = "Dump directory not created"
             raise OpTestError(msg)
-        self.c.run_command("rm -rf /var/crash/%s; sync" % self.crash_content[0])
+        if self.dump_server_user != 'root':
```

Comment:
Can be simplified as follows

        if self.dump_server_user == 'root':
            self.c.run_command("rm -rf /var/crash/%s; sync" % self.crash_content[0])

Reply:
Done! Thanks